### PR TITLE
hotfix: check if the identity from the SSO is still valid, and regenerat…

### PR DESCRIPTION
**Pull request by kuruk-mm**
…e it if not

## What does this PR change?

Validates if the Identity from the Single Sign On is still valid, and if not, regenerate it.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### <samp>🤖 Generated by Copilot at e3d4bab</samp>

Improved auth chain management in `ssoIdentity` validation. Added a check in `session/sagas.ts` to reuse valid auth chains and avoid creating new ones unnecessarily.

### CODE REVIEW

File: browser-interface/packages/shared/session/sagas.ts [hunk 1]
```diff
@@ -321,7 +321,9 @@ async function createAuthIdentity(requestManager: RequestManager, isGuest: boole
 
   const ssoIdentity = await SingleSignOn.getIdentity(address)
 
-  if (!ssoIdentity) {
+  const validIdentity = !ssoIdentity ? false : isSessionExpired({ identity: ssoIdentity })
+
+  if (!ssoIdentity || validIdentity) {
```
Comment (fzavalia) [inline browser-interface/packages/shared/session/sagas.ts:326]:
```suggestion
  if (validIdentity) {
```
This should be enough I believe

File: browser-interface/packages/shared/session/sagas.ts [hunk 2]
```diff
@@ -321,7 +321,9 @@ async function createAuthIdentity(requestManager: RequestManager, isGuest: boole
 
   const ssoIdentity = await SingleSignOn.getIdentity(address)
 
-  if (!ssoIdentity) {
+  const validIdentity = !ssoIdentity ? false : isSessionExpired({ identity: ssoIdentity })
```
Comment (fzavalia) [inline browser-interface/packages/shared/session/sagas.ts]: Maybe I'm wrong, but this logic would mean:
- If no identity is obtained, it is NOT valid
- If the identity is expired, validIdentity would be TRUE
The last one should be the other way around right?

Comment (kuruk-mm) [reply browser-interface/packages/shared/session/sagas.ts]: You're right. I have just fixed it, the logic was wrong

File: browser-interface/packages/shared/session/index.ts [hunk 3]
```diff
@@ -57,7 +58,9 @@ async function getSessionWithSSOIdentity(sessions: StoredSession[]) {
 
   const ssoIdentity = await SingleSignOn.getIdentity(session.identity.address)
 
-  if (!ssoIdentity) {
+  const expiredIdentity = ssoIdentity ? isSessionExpired({ identity: ssoIdentity }) : true
+
+  if (!ssoIdentity || expiredIdentity) {
```
Comment (fzavalia) [inline browser-interface/packages/shared/session/index.ts:63]:
```suggestion
  if (!ssoIdentity || isSessionExpired({ identity: ssoIdentity })) {
```
Maybe this is simpler

File: browser-interface/packages/shared/session/sagas.ts [hunk 4]
```diff
@@ -321,7 +321,9 @@ async function createAuthIdentity(requestManager: RequestManager, isGuest: boole
 
   const ssoIdentity = await SingleSignOn.getIdentity(address)
 
-  if (!ssoIdentity) {
+  const expiredIdentity = ssoIdentity ? isSessionExpired({ identity: ssoIdentity }) : true
+
+  if (!ssoIdentity || expiredIdentity) {
```
Comment (fzavalia) [inline browser-interface/packages/shared/session/sagas.ts:326]:
```suggestion
  if (!ssoIdentity || isSessionExpired({ identity: ssoIdentity })) {
```
Maybe this is simpler

**Review by fzavalia — APPROVED** [review]

**Pull request merged**